packages/mdc-textfield/input/README.md

```diff
@@ -0,0 +1,66 @@
+<!--docs:
+title: "Text Field Input"
+layout: detail
+section: components
+excerpt: "The input element is where the user enters text"
+iconId: input
+path: /catalog/input-controls/text-fields/input/
+-->
+
+# Text Field Input
+
+The input element is where the user enters text.
+
+## Design & API Documentation
+
+<ul class="icon-list">
+  <li class="icon-list-item icon-list-item--spec">
+    <a href="https://material.io/guidelines/components/text-fields.html#text-fields-field-types">Material Design guidelines: Text Field Types</a>
+  </li>
+</ul>
+
+
+## Usage
+
+#### MDCTextFieldInput API
+
+##### MDCTextFieldInput.foundation
+
+MDCTextFieldInputFoundation. This allows the parent MDCTextField component to access the public methods on the MDCTextFieldInputFoundation class.
+
+### Using the foundation class
+
+Method Signature | Description
+--- | ---
+`registerEventHandler(evtType: string, handler: EventListener) => void` | Registers an event listener on the input element for a given event
+`deregisterEventHandler(evtType: string, handler: EventListener) => void` | Deregisters an event listener on the input element for a given event
+`getNativeInput() => {value: string, disabled: boolean, badInput: boolean, checkValidity: () => boolean}?` | Returns an object representing the native text input element, with a similar API shape. The object returned should include the `value`, `disabled` and `badInput` properties, as well as the `checkValidity()` function. We _never_ alter the value within our code, however we _do_ update the disabled property, so if you choose to duck-type the return value for this method in your implementation it's important to keep this in mind. Also note that this method can return null, which the foundation will handle gracefully.
+`notifyFocusAction() => void` | Emits a custom event "MDCTextFieldInput:focus" denoting the input is focused
+`notifyBlurAction() => void` | Emits a custom event "MDCTextFieldInput:blur" denoting the input is blurred
+`notifyPressedAction() => void` | Emits a custom event "MDCTextFieldInput:pressed" denoting the input is pressed
+
+#### The full foundation API
+
+##### MDCTextFieldInputFoundation.isBadInput() => boolean
+
+True if the input fails validity checks.
+
+##### MDCTextFieldInputFoundation.getValue() => string
+
+Returns the value in the input.
+
+##### MDCTextFieldInputFoundation.checkValidity() => boolean
+
+Returns the result of the checkValidity() call in the native input. This can be the result of customized validity checking.
```

Comment:
This isn't true, is it? The `checkValidity` foundation method simply results in calling `checkValidity` directly on the native input element. (Don't we already have an existing issue related to there currently being no way to retrieve custom validity set by `setValid`?)

Reply:
Ya I think you're right Ken. I guess we don't even reference custom validity in the input subelement foundation. We only reference custom validity in the parent MDCTextFieldFoundation. So we should drop this from the README.

Sorry for the confusion Bonnie, but revert this back to how it was.